src/Events/Controllers/AppEventsController.cs

```diff
@@ -217,6 +211,31 @@ public async Task<ActionResult<string>> Post([FromBody] CloudEventRequestModel c
             }
         }
 
+        private static (bool IsValid, string ErrorMessage) ValidateQueryParams(string after, DateTime? from, DateTime? to, int size)
+        {
+            if (string.IsNullOrEmpty(after) && from == null)
+            {
+                return (false, "From or after must be defined.");
+            }
+
+            if (from != null && from.Value.Kind == DateTimeKind.Unspecified)
+            {
+                return (false, "From must specify timezone. E.g. 2022-07-07T11:00:53.3917Z for UTC");
+            }
+
+            if (to != null && to.Value.Kind == DateTimeKind.Unspecified)
+            {
+                return (false, "To must specify timezone. E.g. 2022-07-07T11:00:53.3917Z for UTC");
```

Comment:
```suggestion
                return (false, "The 'To' parameter must specify timezone. E.g. 2022-07-07T11:00:53.3917Z for UTC");
```
Same suggestion for the other messages. It took me a few seconds before I understood the sentence the way it was.